### PR TITLE
Add Tensornet2 architecture

### PR DESCRIPTION
**Pull request by sef43**
- adds the new tensornet2 architecture for the aceff-2.0 pre-trained model.

- the key addition is Aimnet2 style neutral charge equilibration and a coulomb interaction.

- This PR adds the tensonet2 model and a new output model that does a coulomb interaction using the predicted charges.

Note that the parent branch is #373

### CODE REVIEW

**Comment by sef43** [comment]
It also includes some Tensornet optimizations first attempted in this PR: https://github.com/torchmd/torchmd-net/pull/320
which reduce memory by ~30% and increase speed by ~30%, where possible:
- I is kept as a scalar with shape [...,F]
- A is kept as a vector with shape [...,F,3]
- S is still a tensor with shape [...,F,3,3]

where F is the embedding dimension and the leading dim is the number of nodes or number of edges.
